### ISSUE TITLE
Denial of service caused by malformed messages

### ISSUE THREAD

**Issue by henokgetachew** [medic/medic-webapp#4101]
Placeholder for https://github.com/medic/medic-projects/issues/3248

**Comment by sglangevin** [comment]
@vimemo what is the status of this issue? I'd like to get everything in 2.15 closed off soon if we can.

**Comment by vimemo** [comment]
@sglangevin, the latest commit should take care of this issue but we are optimizing the messages view with some recommendations from @garethbowen. The optimization depends on #3826 which is currently in code review. I should have this ready for code review right after that; hopefully eod.

**Comment by sglangevin** [comment]
Cool - thanks for the update. Are we going to backport #3826 to 2.15.x? If so, we should add that ticket to the 2.15.0 project so that we make sure it gets tested as well.

**Comment by garethbowen** [comment]
I've created a new issue (#4284) for the refactor that @vimemo is referring to. The above PR can be merged as is to provide a temporary workaround for 2.15.0 as we don't want to include large refactors this late in the 2.15.0 release cycle.

**Comment by garethbowen** [comment]
Merged in https://github.com/medic/medic-webapp/commit/524f92386d7fe6f6c177aa435f9dc0cae492dc0a

**Comment by henokgetachew** [comment]
Can we get this backported to 2.14 as well?
